lib/srv/termhandlers.go

```diff
@@ -123,11 +123,20 @@ func (t *TermHandlers) HandleShell(ctx context.Context, ch ssh.Channel, req *ssh
 	if err := scx.SetExecRequest(execRequest); err != nil {
 		return trace.Wrap(err)
 	}
-	if err := t.SessionRegistry.OpenSession(ctx, ch, scx); err != nil {
-		return trace.Wrap(err)
+
+	if joinID, joinMode := scx.GetJoinParams(); joinID != "" {
+		err := t.SessionRegistry.JoinSession(ctx, ch, scx, joinID, joinMode)
+
+		// TODO(Joerger): DELETE IN v19.0.0 - old clients set TELEPORT_SESSION for new
+		// sessions, but the client should only provide a session ID for join sessions.
+		// Therefore we ignore the client's provided session ID if there is no session
+		// to join and overwrite the TELEPORT_SESSION env var for the session.
```

Comment:
@capnspacehook I gave this some thought and tested with a v17 client connecting to a server with this PR based off v18, and it seemed to work fine. I don't think the client provided session ID needs to be respected now that the Node will always set it itself (if a join session isn't found). WDYT?

Reply:
Isn't this the current behavior, just moved?

We might want to actually respect the DELETE IN and return the error in master and add the escape hatch for `IsNotFound` in the backport tho, or maybe bump the DELETE IN to v20 just to be extra accomodating to older clients.